src/validation.cpp

```diff
@@ -687,6 +687,40 @@ bool MemPoolAccept::PreChecks(ATMPArgs& args, Workspace& ws)
         }
     }
 
+    for (const CTxIn& txin : tx.vin)
+    {
+        std::vector<std::vector<unsigned char>> stack;
+        auto& pocketTx = ws.m_pocketTx;
+
+        if (!EvalScript(stack, txin.scriptSig, SCRIPT_VERIFY_NONE, BaseSignatureChecker(), SigVersion::BASE))
+            return state.ConsensusFailed(TxValidationResult::TX_SOCIAL_CONSENSUS, strprintf("Failed to get scripSig stack!"));
+        unsigned int len = stack.size();
+        if (len == 4 && CScript::EncodeOP_N(stack[len -1][0]) == OP_TRUE && stack[len-2][0] == 'a' && stack[len-2][1] == 'd')
+        {
+            PocketBlockRef adBlock = PocketDb::TransRepoInst.List(std::vector<std::string>{txin.prevout.hash.ToString()}, true, false, false);
```

Comment:
It is good to perform additional checks here. And better to use `Get` instead of `List` because it already checks for empty array. However, transaction can still be null and we for sure need to check this. Also neither `List` nor `Get` guarantee that any of fields of transaction will be filled except type and time, so probably check for them too (this is not that critical as the check for null though).